src/ir_Transcold.h

```diff
@@ -0,0 +1,187 @@
+temp 	29	Auto	cool	close (right)	111011110001000001101100100100010101010010101011
+temp 	28	Auto	cool	close (right)	111011110001000001100010100100010101010010101011
+temp 	27	Auto	cool	close (right)	111011110001000001101010100100010101010010101011
+temp 	26	Auto	cool	close (right)	111011110001000001100110100100010101010010101011
+temp 	25	Auto	cool	close (right)	111011110001000001101110100100010101010010101011
+temp 	24	Auto	cool	close (right)	111011110001000001100001100111100101010010101011
+temp 	23	Auto	cool	close (right)	111011110001000001101001100101100101010010101011
+temp 	22	Auto	cool	close (right)	111011110001000001100101100101100101010010101011
+temp 	21	Auto	cool	close (right)	111011110001000001101101100101100101010010101011
+temp 	20	Auto	cool	close (right)	111011110001000001100011100101100101010010101011
+temp 	19	Auto	cool	close (right)	111011110001000001101011100101100101010010101011
+temp 	18	Auto	cool	close (right)	111011110001000001100111100110000101010010101011
+temp 	17	Auto	cool	close (right)	111011110001000001100111100110000101010010101011
+temp 	16	Auto	cool	close (right)	111011110001000001100111100110000101010010101011
+
+ **************************************************************************************************************/
+
+#ifndef IR_TRANSCOLD_H_
+#define IR_TRANSCOLD_H_
+
+#define __STDC_LIMIT_MACROS
+#include <stdint.h>
+#ifndef UNIT_TEST
+#include <Arduino.h>
+#endif
+#include "IRremoteESP8266.h"
+#include "IRsend.h"
+#ifdef UNIT_TEST
+#include "IRsend_test.h"
+#endif
+
+// Constants
+// Modes
+const uint8_t kTranscoldCool = 0b0110;
+const uint8_t kTranscoldDry = 0b1100;
+const uint8_t kTranscoldAuto = 0b1110;
+const uint8_t kTranscoldHeat = 0b1010;
+const uint8_t kTranscoldFan = 0b0010;
+const uint8_t kTranscoldModeOffset = 12;
+const uint8_t kTranscoldModeSize = 4;
+
+// Fan Control
+const uint8_t kTranscoldFanOffset = 16;
+const uint8_t kTranscoldFanSize = 4;
+const uint8_t kTranscoldFanMin = 0b1001;
+const uint8_t kTranscoldFanMed = 0b1101;
+const uint8_t kTranscoldFanMax = 0b1011;
+const uint8_t kTranscoldFanAuto = 0b1111;
+const uint8_t kTranscoldFanAuto0 = 0b0110;
+const uint8_t kTranscoldFanZoneFollow = 0b0000;
+const uint8_t kTranscoldFanFixed = 0b1100;
+
+// Temperature
+const uint8_t kTranscoldTempMin = 17;  // Celsius
+const uint8_t kTranscoldTempMax = 30;  // Celsius
+const uint8_t kTranscoldTempRange = kTranscoldTempMax - kTranscoldTempMin + 1;
+const uint8_t kTranscoldFanTempCode = 0b1111;  // Part of Fan Mode.
+const uint8_t kTranscoldTempOffset = 8;
+const uint8_t kTranscoldTempSize = 4;
+const uint8_t kTranscoldTempMap[kTranscoldTempRange] = {
+    0b0111,  // 17C transcold starts with 18
+    0b0111,  // 18c
+    0b1011,  // 19C
+    0b0011,  // 20C
+    0b1101,  // 21C
+    0b0101,  // 22C
+    0b1001,  // 23C
+    0b0001,  // 24C
+    0b1110,  // 25C
+    0b0110,  // 26C
+    0b1010,  // 27C
+    0b0010,  // 28C
+    0b1100,  // 29C
+    0b0100   // 30C
+};
```

Comment:
Hmm, is min temp actually 18 and not 17? is there really no logic to this madness?

Reply:
@iamDshetty What's the story here with this comment/code? What is the actual min temp? Why is 17C the same as 18C? etc etc?

Reply:
@crankyoldgit @NiKiZe 
Min temp of Transcold AC is 18 deg C.
We have a general UI for all AC's which starts from 17deg C. So In general 17C is required for us. So kept same as 18C as there is no 17C for this AC. 

The AC which is with me is an Old model of Transcold, doesn't mean there is no new brand of Transcold. Transcold brand is still available with new Models.

I have shared the bit calculation sheet. You can create a new file for this AC with min file size.

I will definitely help you out with Testing.

Reply:
I've redone how temperature is encoded/decoded. I've replaced the lookup-table with bit manipulation and math to make it safer and faster.

It now uses a min temp of 18C too.

Reply:
@NiKiZe You should probably re-review that change/part etc.